R/lint.R

```diff
@@ -125,64 +125,53 @@ reorder_lints <- function(lints) {
     ]
 }
 
-#' Lint a package
+#' Lint a directory
 #'
-#' Apply one or more linters to all of the R files in a package.
-#' @param path the path to the base directory of the package, if \code{NULL},
+#' Apply one or more linters to all of the R files in a directory
+#' @param path the path to the base directory, if \code{NULL},
 #' it will be searched in the parent directories of the current directory.
 #' @param relative_path if \code{TRUE}, file paths are printed using their path
-#' relative to the package base directory.  If \code{FALSE}, use the full
+#' relative to the base directory.  If \code{FALSE}, use the full
 #' absolute path.
 #' @param ... additional arguments passed to \code{\link{lint}}, e.g.
 #' \code{cache} or \code{linters}.
 #' @param exclusions exclusions for \code{\link{exclude}}, relative to the
 #' package path.
+#' @param pattern pattern for files, by default it will take files with .R or .r extension.
 #' @inherit lint_file return
 #' @examples
 #' \dontrun{
-#'   lint_package()
-#'   lint_package(
+#'   lint_dir()
+#'   lint_dir(
 #'     linters = list(semicolon_terminator_linter())
 #'     cache = TRUE,
 #'     exclusions = list("inst/doc/creating_linters.R" = 1, "inst/example/bad.R")
 #'   )
 #' }
 #' @export
-lint_package <- function(path = ".", relative_path = TRUE, ..., exclusions = NULL) {
-  path <- find_package(path)
-
-  read_settings(path)
-  on.exit(clear_settings, add = TRUE)
-
-  exclusions <- normalize_exclusions(c(exclusions, settings$exclusions), FALSE)
-
-  files <- dir(
-    path = file.path(path,
-                     c("R",
-                       "tests",
-                       "inst")
-                     ),
-    pattern = rex::rex(".", one_of("Rr"), end),
+lint_dir <- function(path = ".", relative_path = TRUE, ..., exclusions = NULL, pattern = rex::rex(".", one_of("Rr"), end)) {
+  files <- dir( path,
+    pattern = pattern,
     recursive = TRUE,
     full.names = TRUE
   )
 
   # Remove fully ignored files to avoid reading & parsing
   to_exclude <- vapply(seq_len(length(files)),
-    function(i) {
-      file <- files[i]
-      file %in% names(exclusions) && exclusions[[file]] == Inf
-     },
-    logical(1))
+                       function(i) {
+                         file <- files[i]
+                         file %in% names(exclusions) && exclusions[[file]] == Inf
```

Comment:
`exclusions` is no longer defined within the scope of this function. So this should throw an error. Did you add any tests for `lint_dir`, or is `lint_dir` only tested through a call from `lint_package` (where `exclusions` is defined)?